### PR TITLE
Add dcap_quoteverify FFI

### PR DESCRIPTION
**Pull request by nick-mobilecoin**
Adds FFI to the sgx functionality provided in
`libsgx_dcap_quoteverify.so`

### CODE REVIEW

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #53** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/53" target="_blank">Graphite</a> 
    * **PR #55** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/55" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/55?utm_source=stack-comment).

**Review by samdealy — APPROVED** [review]

File: dcap_quoteverify/sys/build.rs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+const DCAP_QL_FUNCTIONS: &[&str] = &[
+    "sgx_qv_free_qve_identity",
+    "sgx_qv_get_quote_supplemental_data_size",
+    "sgx_qv_get_qve_identity",
+    "sgx_qvl_get_quote_supplemental_data_size",
+    "sgx_qvl_get_quote_supplemental_data_version",
+    "sgx_qvl_verify_quote",
+    "sgx_qv_set_enclave_load_policy",
+    "sgx_qv_set_path",
+    "sgx_qv_set_qpl_path",
```
Comment (awygle) [inline dcap_quoteverify/sys/build.rs:16]: `sgx_qv_set_qpl_path` and `sgx_qv_set_qve_path` are both defined in `sgx_dcap_pcs_com.h`, which isn't listed in `wrapper.h`. That header looks like a public header so I think it should be added to `wrapper.h`.

Separately, that header also defines a number of other functions and several function pointer types which should probably be bound as well.

**Review by awygle — CHANGES_REQUESTED** [review]
I am not especially confident that I properly understand the situation with `sgx_dcap_pcs_com.h` in particular so I'm very open to hearing that I've misinterpreted things with some of this review.

File: dcap_quoteverify/sys/build.rs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+//! Builds the FFI function bindings for dcap quoteverify library of the Intel
+//! SGX SDK
+
+use cargo_emit::rustc_link_lib;
+
+const DCAP_QL_FUNCTIONS: &[&str] = &[
+    "sgx_qv_free_qve_identity",
+    "sgx_qv_get_quote_supplemental_data_size",
+    "sgx_qv_get_qve_identity",
+    "sgx_qvl_get_quote_supplemental_data_size",
```
[the remark this answers is not in the document]
Comment (nick-mobilecoin) [reply dcap_quoteverify/sys/build.rs]: good catch!  Those should not be bound

File: dcap_quoteverify/sys/types/build.rs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+const DCAP_QUOTEVERIFY_TYPES: &[&str] = &[
+    "sgx_qv_path_type_t",
+    "_sgx_ql_qv_result_t",
+    "_sgx_ql_qv_supplemental_t",
+    "_pck_cert_flag_enum_t",
+    "_sgx_ql_att_key_id_param_t",
+    "_sgx_ql_att_id_list_t",
+    "_sgx_ql_qe_report_info_t",
+    "_quote_nonce",
```
[the remark this answers is not in the document]
Comment (nick-mobilecoin) [reply dcap_quoteverify/sys/types/build.rs]: another good catch, I got blinded by the quote name and failed to realize that was in the sgx sdk and not the dcap libraries.

File: dcap_quoteverify/sys/types/build.rs [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    "_sgx_ql_att_key_id_list_header_t",
```
[the remark this answers is not in the document]
Comment (nick-mobilecoin) [reply dcap_quoteverify/sys/types/build.rs:13]: added the others from quote3

File: dcap_quoteverify/sys/types/build.rs [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    "_sgx_att_key_id_ext_t",
```
[the remark this answers is not in the document]
Comment (nick-mobilecoin) [reply dcap_quoteverify/sys/types/build.rs]: done

File: dcap_quoteverify/sys/build.rs (hunk 1 above)
Comment (nick-mobilecoin) [reply dcap_quoteverify/sys/build.rs]: removed `sgx_qv_set_qpl_path`.  that is accessed internally via `sgx_qv_set_path`.  The `sgx_dcap_pcs_com.h` looks to defining functions for the `sgx_dcap_quoteprov` library.  There is one provided with the DCAP build, or one can choose to provide their own implementation, but these are all accessed via the DCAP c libraries and shouldn't need to be exposed to the rust side of things at all.

**Review by awygle — APPROVED** [review]
LGTM, thanks!

**Pull request merged**